### ISSUE TITLE
SpaceAfterPropertyColon crashes on property names spanning multiple lines

### ISSUE THREAD

**Issue by rvanlieshout** [sds/scss-lint#824]
Hi,

I've encountered an issue with sass-lint in #offset_to. This method seems to fail on

``` ruby
actual_index && actual_index + 1 - source_position.offset
```

where:
- source_position.inspect: '514:17' 
- pattern: ':'
- actual_line: 513
- engine.lines.size: 640
- actual_index: ''
- actual_offset: 16

source around that actual_line (full source further down the line):

``` scss
509 .dataTables_wrapper .popover {
510   max-width: 100%;
511   width: 700px;
512 
513   .row.filter {
514     width: 100%;*
515     zoom: 1;
516     margin: 0px;
517   }
```

I don't even know why that \* there is in place, but it should not fail this hard just to let me know :)
# Stack

```
SCSSLint::Linter::SpaceAfterPropertyColon raised unexpected error linting file /home/pronto/projects/kiss/opvangapp/app/assets/stylesheets/partials/base.scss: 'undefined method `+' for nil:NilClass'
/usr/local/share/gems/gems/scss_lint-0.49.0/lib/scss_lint/linter/space_after_property_colon.rb:80:in `whitespace_after_colon'
/usr/local/share/gems/gems/scss_lint-0.49.0/lib/scss_lint/linter/space_after_property_colon.rb:16:in `visit_prop'
/usr/local/share/gems/gems/sass-3.4.21/lib/sass/tree/visitors/base.rb:36:in `visit'
/usr/local/share/gems/gems/scss_lint-0.49.0/lib/scss_lint/linter.rb:138:in `visit'
/usr/local/share/gems/gems/scss_lint-0.49.0/lib/scss_lint/linter.rb:149:in `block in visit_children'
/usr/local/share/gems/gems/scss_lint-0.49.0/lib/scss_lint/linter.rb:147:in `each'
/usr/local/share/gems/gems/scss_lint-0.49.0/lib/scss_lint/linter.rb:147:in `visit_children'
/usr/local/share/gems/gems/sass-3.4.21/lib/sass/tree/visitors/base.rb:36:in `block in visit'
/usr/local/share/gems/gems/scss_lint-0.49.0/lib/scss_lint/linter/space_after_property_colon.rb:12:in `visit_rule'
/usr/local/share/gems/gems/sass-3.4.21/lib/sass/tree/visitors/base.rb:36:in `visit'
/usr/local/share/gems/gems/scss_lint-0.49.0/lib/scss_lint/linter.rb:138:in `visit'
/usr/local/share/gems/gems/scss_lint-0.49.0/lib/scss_lint/linter.rb:149:in `block in visit_children'
/usr/local/share/gems/gems/scss_lint-0.49.0/lib/scss_lint/linter.rb:147:in `each'
/usr/local/share/gems/gems/scss_lint-0.49.0/lib/scss_lint/linter.rb:147:in `visit_children'
/usr/local/share/gems/gems/sass-3.4.21/lib/sass/tree/visitors/base.rb:36:in `block in visit'
/usr/local/share/gems/gems/scss_lint-0.49.0/lib/scss_lint/linter/space_after_property_colon.rb:12:in `visit_rule'
/usr/local/share/gems/gems/sass-3.4.21/lib/sass/tree/visitors/base.rb:36:in `visit'
/usr/local/share/gems/gems/scss_lint-0.49.0/lib/scss_lint/linter.rb:138:in `visit'
/usr/local/share/gems/gems/scss_lint-0.49.0/lib/scss_lint/linter.rb:149:in `block in visit_children'
/usr/local/share/gems/gems/scss_lint-0.49.0/lib/scss_lint/linter.rb:147:in `each'
/usr/local/share/gems/gems/scss_lint-0.49.0/lib/scss_lint/linter.rb:147:in `visit_children'
/usr/local/share/gems/gems/sass-3.4.21/lib/sass/tree/visitors/base.rb:38:in `visit'
/usr/local/share/gems/gems/scss_lint-0.49.0/lib/scss_lint/linter.rb:138:in `visit'
/usr/local/share/gems/gems/scss_lint-0.49.0/lib/scss_lint/linter.rb:41:in `run'
/usr/local/share/gems/gems/scss_lint-0.49.0/lib/scss_lint/runner.rb:53:in `run_linter'
/usr/local/share/gems/gems/scss_lint-0.49.0/lib/scss_lint/runner.rb:35:in `block in find_lints'
/usr/local/share/gems/gems/scss_lint-0.49.0/lib/scss_lint/runner.rb:33:in `each'
/usr/local/share/gems/gems/scss_lint-0.49.0/lib/scss_lint/runner.rb:33:in `find_lints'
/usr/local/share/gems/gems/scss_lint-0.49.0/lib/scss_lint/runner.rb:19:in `block in run'
/usr/local/share/gems/gems/scss_lint-0.49.0/lib/scss_lint/runner.rb:18:in `each'
/usr/local/share/gems/gems/scss_lint-0.49.0/lib/scss_lint/runner.rb:18:in `run'
/usr/local/share/gems/gems/scss_lint-0.49.0/lib/scss_lint/cli.rb:74:in `scan_for_lints'
/usr/local/share/gems/gems/scss_lint-0.49.0/lib/scss_lint/cli.rb:59:in `act_on_options'
/usr/local/share/gems/gems/scss_lint-0.49.0/lib/scss_lint/cli.rb:33:in `run'
/usr/local/share/gems/gems/scss_lint-0.49.0/bin/scss-lint:7:in `<top (required)>'
/usr/local/bin/scss-lint:23:in `load'
/usr/local/bin/scss-lint:23:in `<main>'
```
# Versions
- SCSS-Lint version: 0.49.0
- Sass version: 3.4.21
- Ruby version: 2.0.0
# .scss-lint.yml

``` yam
linters:
  SingleLinePerSelector:
    enabled: false
  QualifyingElement:
    enabled: false
```
# File

``` scss
@mixin header_gradient {
  background-color: #fafafa;
  background-image: -moz-linear-gradient(top, #ffffff, #f2f2f2);
  background-image: -webkit-gradient(linear, 0 0, 0 100%, from(#ffffff), to(#f2f2f2));
  background-image: -webkit-linear-gradient(top, #ffffff, #f2f2f2);
  background-image: -o-linear-gradient(top, #ffffff, #f2f2f2);
  background-image: linear-gradient(to bottom, #ffffff, #f2f2f2);
  background-repeat: repeat-x;
  filter: progid:DXImageTransform.Microsoft.gradient(startColorstr='#ffffffff', endColorstr='#fff2f2f2', GradientType=0);
  border-bottom: 1px solid #ddd;
}

.ajax-loader {
  position: absolute;
  z-index: 1;
  color: white;
  top: 15px;
  left: 115px;
}

.select2-container {
  width: 210px;

  &.input-mini {
    width: 70px;
  }
  &.input-small {
    width: 110px;
  }
  &.input-medium {
    width: 150px;
  }
  &.input-large {
    width: 210px;
  }
  &.input-xlarge {
    width: 270px;
  }
  &.input-xxlarge {
    width: 530px;
  }
}

.title {
  form {
    margin: 0px;
  }
  .select2-container a {
    font-weight: normal;
    font-size: 14px;
  }
  .pagination {
    margin: 5px 0 0 0;
    font-weight: normal;
    font-size: 14px;
  }
}

html, body {
  min-height: 100%;
}

.print-only {
  visibility: hidden;
  height: 0;
}

.table th {
  white-space: nowrap;
}
.table td {
  &.boolean {
    text-align: center
  }
  &.edit {
    text-align: right;
  }
}
.table td, .table th {
  &.boolean_childminding_agency,
  &.boolean_daycare,
  &.boolean_oshc {
    text-align: center;
  }

  &.edit {
    width: 25px;
  }

  &.boolean {
    width: 70px;
  }

  &.verplicht,
  &.mandatory,
  &.kvk {
    width: 80px;
  }

  &.verplicht {
    text-align: center;
  }

  &.postal_code,
  &.combinations {
    width: 100px;
  }

  &.date_of_expiration,
  &.date_of_expiration_mandatory,
  &.date_of_issue,
  &.date_of_birth {
    width: 120px;
  }

  &.document_percentage,
  &.residence {
    width: 200px;
  }

  &.download {
    width: 400px;
  }

  &.name {
    width: 20%;
  }
}

.main-container {
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;

  .title {
    @include header_gradient;
    color: #555;
    padding: 0 15px;

    h1 {
      font-size: 20px;
      font-weight: normal;
    }
  }
  .navbar {
    margin-bottom: 0;
  }

  .navbar .nav > .active > a {
    color: #333;
  }

  .navbar.secondary {
    .navbar-inner {
      min-height: 30px;
      a {
        padding: 5px 10px 5px;
        font-size: 11px;
      }
    }
  }

  .navbar-inner {
    padding: 0;
    border-left: none;
    border-right: none;
    border-radius: 0;
    border-top: 0px;
  }

  .tab-content, .page-content {
    padding: 20px 20px 0px 20px;
  }

  .tab-pane {
    h2 {
      display: none;
    }
  }
}

body.wizard .main-container {
  margin-bottom: 1em;

  form {
    margin-bottom: 0px;
  }

  .page-actions {
    .right {
      text-align: right;
    }
  }
  .combined_invoices {
    margin-top: 18px;
  }
}

.form-horizontal, .form-actions {
  margin-bottom: 0px;
}
.form-actions {
  background-color: transparent;
}

.page-actions {
  margin-top: 20px;
  margin-bottom: 20px;
  *zoom: 1;
  *zoom: 1;
}
.page-actions:before,
.page-actions:after {
  display: table;
  content: "";
  line-height: 0;
}
.page-actions:after {
  clear: both;
}

.dataTables_wrapper {
  margin: 15px 0px;
  padding: 0px;
  border: 1px solid #ddd;
  -webkit-border-radius: 4px;
  -moz-border-radius: 4px;
  border-radius: 4px;

  .row-fluid {
    @include header_gradient;

    padding-top: 6px;
    border-bottom: 0px;
    color: #888;

    height: 35px;
    overflow: hidden;

    .span4 label {
      position: relative;
      top: -5px;
    }
    .dataTables_filter label {
      float: right;
      margin-left: 5px;
      &.end-date-label {
        float: left;
      }
    }
  }

  table.dataTable {
    border-top: 1px solid #ddd;
    border-bottom: 1px solid #ddd;
    margin-bottom: 0px !important;

    background-color: white;
    opacity: 1;
    // transition: all 0.25s ease;
    // -moz-transition: all 0.25s ease;
    // -webkit-transition: all 0.25s ease;

    &.loading {
      opacity: 0.3;
      // background-color: #F0F0F0;
    }
   }

  div.dataTables_length label {
    margin-left: 10px;
    margin-top: 5px;
    select {
      margin-bottom: 4px;
      opacity: 0.8;
      padding: 0px;
      line-height: 20px;
    }
  }
  div.dataTables_filter {
    height: 35px;
    label {
      margin-bottom: 0px;
      margin-right: 3px;
    }
    input {
      margin-bottom: 0px;
      opacity: 0.8;
    }
  }
  table.dataTable {
  }
  div.dataTables_info {
    padding-top: 4px;
    margin-left: 10px;
  }
  div.dataTables_paginate {
    margin-right: 2px;
  }
}

.period_selector {
  display: none;
}
.dataTables_length .select2-container {
  width: 80px;
}

div.dataTables_filter {
  input {
    width: 120px;
  }
  .period_selector {
    display: inline;
    float: right;
    margin-right: 6px;
    opacity: 0.8;
    white-space: nowrap;
    select, input, .select2-container {
      width: 90px;
    }
    label {
      float: none;
      display: inline;
      position: relative;
      top: 5px;
      margin-left: 5px;
    }
    &#childminder_registration_report_datatable_period_selector, &#nursery_registration_report_datatable_period_selector {
      .btn {
        position: relative;
        top: -5px;
      }
    }

    &#price_list_legend {
      .btn {
        top: 0px;
      }
    }

    &#price_list_legend {
      .btn {
        top: 0px;
      }
    }
  }
}

.DataTables_sort_icon.css_right.ui-icon {
  font-family: FontAwesome;
  font-weight: normal;
  font-style: normal;
  text-decoration: inherit;
  display: inline;
  width: auto;
  height: auto;
  line-height: normal;
  vertical-align: baseline;
  background-image: none !important;
  background-position: 0% 0%;
  background-repeat: repeat;
  text-decoration: inherit;
  display: inline-block;
  speak: none;
  float: left;
  margin-right: 5px;
  position: relative;
  top: 3px;
}

.ui-icon-carat-2-n-s:before   { content: "\f0dc"; color: silver; }
.ui-icon-triangle-1-s:before  { content: "\f0dd"; color: black; }
.ui-icon-triangle-1-n:before  { content: "\f0de"; color: black; }

@media (min-width: 768px) {
  .select2-drop {
    max-width: 600px;
  }
}
@media (max-width: 767px) {
  .select2-drop {
    // max-width: 400px;
    width: auto !important;
    margin-left: 20px;
    margin-right: 20px;
    left: 0px !important;
  }
}

div#documents_table {
  padding: 0px;

  border: 1px solid #ddd;
  -webkit-border-radius: 4px;
  -moz-border-radius: 4px;
  border-radius: 4px;

  margin-bottom: 1em;

  table {
    width: 100%;
    margin: 0px;

    thead {
      th {
        @include header_gradient;

        padding-top: 6px;
        border-bottom: 0px;
        color: black;

        overflow: hidden;
      }
    }

    td.indent {
      width: 10px;
    }

    td {
      input, label {
        margin: 0px;
      }
    }

    td.date-of-expiration { width: 80px; }
    td.status {
      width: 80px;
      &.new {}
      &.progress {}
      &.completed {}
      &.approved { color: green; }
      &.disapproved { color: red; }
    }
  }
}

form.childminder_registration,
form.nursery_registration {
  input[type='radio'] {
    margin-top: -4px;
  }

  ul.childminding-agency-contracts-children,
  ul.childminding-agency-contracts-childminders,
  ul.care-locations {
    list-style: none;
    li label {
      display: inline;
    }
  }
}

.form-actions-fixed-bottom {
  position: static;
  &.floating {
    position: fixed;
    bottom: 0px;
    left: 0px;
    right: 0px;
    background-color: white;
    margin-left: 20px;
    margin-right: 20px;
    padding-left: 20px;
    padding-right: 20px;
  }
}
.week_schedule.input {
  .select2-container {
    width: 80px;
  }
}

.price_indication {
  th {
    &.year, &.month {
      text-align: right;
    }
  }
  td {
    text-align: right;
    &.child, &.address {
      text-align: left;
    }
    &.daycare {
      border-left: 0px;
    }
  }
  .sub {
    font-size: 8pt;
    font-style: italic;
  }
  tfoot {
    th {
      text-align: right;
    }
  }
}

// Alle tabellen condensed (uit table.less)
.table {
  th,
  td {
    padding: 4px 5px;
  }
}

.dataTables_wrapper .popover {
  max-width: 100%;
  width: 700px;

  .row.filter {
    width: 100%;*
    zoom: 1;
    margin: 0px;
  }

  text-align: left;

  dl.registration_status_dl, dl.contract_status_dl {
    dt {
      font-weight: normal;
    }
  }

  ul.establishments, ul.person_status, ul.contract_status, ul.family_status,
  ul.child_status, ul.childminder_status, ul.childminder_marketplace_status,
  ul.registration_status, ul.invoice_state, ul.invoice_period, ul.parent_status {
    li {
      list-style-type: none;
      input {
        position: relative;
        top: -3px;
        left: -5px;
      }
    }
  }
}

body.documents.report .dataTables_wrapper .popover {
  width: auto;
}

.input-xlarge.uneditable-input {
  border-color: white;
  color: #333;
  cursor: default;
  background-color: white;
  margin-left: 0px;
  padding-left: 0px;
  -webkit-box-shadow: none;
  -moz-box-shadow: none;
  box-shadow: none;
}

.progress-striped .bar-unknown {
  background-color: #CCC;
  filter: progid:DXImageTransform.Microsoft.gradient(startColorstr='#CCCCCC', endColorstr='#DDDDDD', GradientType=0);

  background-image: -webkit-gradient(linear, 0 100%, 100% 0, color-stop(0.25, rgba(255, 255, 255, 0.15)), color-stop(0.25, transparent), color-stop(0.5, transparent), color-stop(0.5, rgba(255, 255, 255, 0.15)), color-stop(0.75, rgba(255, 255, 255, 0.15)), color-stop(0.75, transparent), to(transparent));
  background-image: -webkit-linear-gradient(45deg, rgba(255, 255, 255, 0.15) 25%, transparent 25%, transparent 50%, rgba(255, 255, 255, 0.15) 50%, rgba(255, 255, 255, 0.15) 75%, transparent 75%, transparent);
  background-image: -moz-linear-gradient(45deg, rgba(255, 255, 255, 0.15) 25%, transparent 25%, transparent 50%, rgba(255, 255, 255, 0.15) 50%, rgba(255, 255, 255, 0.15) 75%, transparent 75%, transparent);
  background-image: -o-linear-gradient(45deg, rgba(255, 255, 255, 0.15) 25%, transparent 25%, transparent 50%, rgba(255, 255, 255, 0.15) 50%, rgba(255, 255, 255, 0.15) 75%, transparent 75%, transparent);
  background-image: linear-gradient(45deg, rgba(255, 255, 255, 0.15) 25%, transparent 25%, transparent 50%, rgba(255, 255, 255, 0.15) 50%, rgba(255, 255, 255, 0.15) 75%, transparent 75%, transparent);

  -webkit-transition: all 0.6s ease;
  -moz-transition: all 0.6s ease;
  -o-transition: all 0.6s ease;
  transition: all 0.6s ease;
}

.progress-striped .bar-complete {
  background-color: #EEE;
  filter: progid:DXImageTransform.Microsoft.gradient(startColorstr='#EEEEEE', endColorstr='#CCCCCC', GradientType=0);

  background-image: -webkit-gradient(linear, 0 100%, 100% 0, color-stop(0.25, rgba(255, 255, 255, 0.15)), color-stop(0.25, transparent), color-stop(0.5, transparent), color-stop(0.5, rgba(255, 255, 255, 0.15)), color-stop(0.75, rgba(255, 255, 255, 0.15)), color-stop(0.75, transparent), to(transparent));
  background-image: -webkit-linear-gradient(45deg, rgba(255, 255, 255, 0.15) 25%, transparent 25%, transparent 50%, rgba(255, 255, 255, 0.15) 50%, rgba(255, 255, 255, 0.15) 75%, transparent 75%, transparent);
  background-image: -moz-linear-gradient(45deg, rgba(255, 255, 255, 0.15) 25%, transparent 25%, transparent 50%, rgba(255, 255, 255, 0.15) 50%, rgba(255, 255, 255, 0.15) 75%, transparent 75%, transparent);
  background-image: -o-linear-gradient(45deg, rgba(255, 255, 255, 0.15) 25%, transparent 25%, transparent 50%, rgba(255, 255, 255, 0.15) 50%, rgba(255, 255, 255, 0.15) 75%, transparent 75%, transparent);
  background-image: linear-gradient(45deg, rgba(255, 255, 255, 0.15) 25%, transparent 25%, transparent 50%, rgba(255, 255, 255, 0.15) 50%, rgba(255, 255, 255, 0.15) 75%, transparent 75%, transparent);
}

.btn.export {
  margin-left: 3px;
}

dl.vertical-with-space {
  dd {
    margin-bottom: 1em;
  }
  dt:after {
    content: ':'
  }
}

dl.dl-horizontal.dl-horizontal-wide {
  dt {
    width: 200px;
  }
  dd {
    margin-left: 220px;
  }
}

.boolean_form_control {
  .controls {
    margin-left: 0px;
  }
}

.small_form_control {
  .control-label {
    width: 100px;
  }
  .controls {
    margin-left: 120px;
  }
}

.control-group.daycare_age {
  .control-group {
    display: inline-block;

    .controls {
      margin-left: 0px;
      max-width: 180px;

      input {
        width: 60px;
      }
    }
  }
}

.footnote {
  font-size: 8pt;
  font-style: italic;
  color: #888;
}
```

**Comment by sds** [comment]
Thanks for the report, @rvanlieshout!

This problem can be reproduced by splitting an asterisk-prefixed property across multiple lines, e.g.:

**Trigger Case**
```scss
.class {
  width: 100%;*  // <-- Asterisk should come before `zoom` on next line
  zoom: 1;
}
```

**Workaround**
```scss
.class {
  width: 100%;
  *zoom: 1;      // <-- Asterisk in correct location
}
```

Unfortunately fixing this is not quick due to a fundamental assumption in the `SpaceAfterPropertyColon`'s implementation that property names span a single line. I'm not clear on if Sass itself should be failing to parse in this case, but if this is still valid it's rather odd.

I won't have time to look into this for now. A pull request with a fix would be really appreciated!
